### PR TITLE
add dualstack tests for digitalocean 

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What does this PR do / Why do we need it**:
Adds dualstack tests for digitalocean. 

**Does this PR close any issues?**:
Part of #10005 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -1424,6 +1424,7 @@ type DigitaloceanNodeSpec struct {
 	Size string `json:"size"`
 	// enable backups for the droplet
 	Backups bool `json:"backups"`
+	// DEPRECATED
```
Comment (rastislavs) [inline pkg/api/v1/types.go:1427]: maybe also mention why - IPv6 is enabled automatically based on IP Family of the cluster?

File: pkg/test/dualstack/cloud_providers_dualstack_test.go [hunk 2]
```diff
@@ -176,6 +177,34 @@ func TestCloudClusterIPFamily(t *testing.T) {
...
 			skipHostNetworkPods: true,
 		},
+		{
+			cloudName: "do",
+			osNames: []string{
+				// "centos",
+				// "flatcar",
+				// "rhel",
+				// "sles",
+				"ubuntu", //  others are unsupported in kkp
```
Comment (rastislavs) [inline pkg/test/dualstack/cloud_providers_dualstack_test.go]: then let's remove the others from the commented code

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve ⭐

**Pull request merged**